plotly/offline/offline.py

```diff
@@ -90,6 +75,11 @@ def iplot(figure_or_data, show_link=True, link_text='Export to plot.ly'):
                                 of the chart that will export the chart to
                                 Plotly Cloud or Plotly Enterprise
     link_text (default='Export to plot.ly') -- the text of export link
+    validate (default=True) -- validate that all of the keys in the figure
+                               are valid? omit if your version of plotly.js
+                               has become outdated with your version of
+                               graph_reference.json or if you need to include
+                               extra, unnecessary keys in your figure.
```

Comment:
This sounds like it could potentially be a bad user experience. Might we be better off just forcing users to keep their `plotly.min.js` file updated?

Rationale:

* say there *is* a conflict between your local version and plotly, when you finally want to push up a file... it would break, which is pretty confusing!
* plotly.js tries very hard to provide backwards compatibility, I think we might be better off just allowing these rare instances to raise errors.
* if we do have an out of sync `plotly.min.js`, then we also need to provide an out of sync `plot-schema`, i feel like we're potentially creating tough environments to recreate for debugging.

Would it be feasible to get away from this pattern?

I like how the graph_reference is setup, where we have the following:

* a method to download the *most recent*
* a copy that lives in a users local `~` directory
* a versioned default (for offline users and such)

The thing I *don't* like is that the initial download is frustratingly slow. Is there any way we can provide this without the long import lag?